images/kindnetd/cmd/kindnetd/main.go

```diff
@@ -54,6 +62,30 @@ func main() {
 	if err != nil {
 		panic(err.Error())
 	}
+
+	// override the internal apiserver endpoint to avoid
+	// waiting for kube-proxy to install the services rules.
+	// If the endpoint is not reachable, fallback the internal endpoint
+	controlPlaneEndpoint := os.Getenv("CONTROL_PLANE_ENDPOINT")
+	if controlPlaneEndpoint != "" && probeTCP(controlPlaneEndpoint, probeTCPtimeout) {
```

Comment:
i'm not sure we should bother probing TCP if we have the env, since we know the env is what we want?

Reply:
this is much faster, you don't have to construct the client twice

Reply:
Under what circumstance would you expect the probe to fail and desire falling back?

Reply:
- wacky iptables/nftables distros/users
- control plane endpoint being a loadbalancer
- HA deployment and the control plane endpoint is removed
- ...
networking :shrug:

Reply:
I mean that's going to break other stuff anyhow 🤔 
I remain skeptical that this complexity is desirable here, I wouldn't expect a production CNI to do this.

Reply:
I see CONTROL_PLANE_ENDPOINT as an input variable from the user that we can't trust.
probeTCP is validation, if is invalid it fall-back to the value that we know that works, that is the apiserver service KUBERNETES_HOST:KUBERNETES_PORT

Reply:
and now that I think it, if we don't fall back ,if the cluster restart and the containers change their ips, kindnet will keep trying to connect to the hardcoded value and we'll not be able to connect.

Reply:
The "user" is us.

It should be receiving a nodename:port, NOT IP.

Reply:
> It should be receiving a nodename:port, NOT IP.

oh, yeah, I was thinking it was the external IP, it uses the internalIP that uses dns

Reply:
but I don't trust dns either :upside_down_face:

Reply:
The docker Edns for the container name should be fine.

Reply:
Whereas the fallback is the in cluster VIP from kube-proxy

Reply:
this is an old commit by the way, is not the way it is now